src/Compilers/Core/AnalyzerDriver/AnalyzerManager.cs

```diff
@@ -223,6 +223,20 @@ internal void ClearAnalyzerExceptionHandlers(DiagnosticAnalyzer analyzer)
             }
         }
 
+        internal bool IsSupportedDiagnostic(DiagnosticAnalyzer analyzer, Diagnostic diagnostic, AnalyzerExecutor analyzerExecutor)
+        {
+            var supportedDescriptors = GetSupportedDiagnosticDescriptors(analyzer, analyzerExecutor);
+            foreach (var descriptor in supportedDescriptors)
```

Comment:
I wrote my comment before reading HeeJae's, but mine read as a response to his. I was actually responding to asking myself the same question.